tests/compiletest.rs

```diff
@@ -86,17 +85,20 @@ fn test_config(target: &str, path: &str, mode: Mode, with_dependencies: bool) ->
 
     let mut config = Config {
         target: Some(target.to_owned()),
-        stderr_filters: STDERR.clone(),
-        stdout_filters: STDOUT.clone(),
-        mode,
         program,
         out_dir: PathBuf::from(std::env::var_os("CARGO_TARGET_DIR").unwrap()).join("ui"),
-        edition: Some("2021".into()), // keep in sync with `./miri run`
         threads: std::env::var("MIRI_TEST_THREADS")
             .ok()
             .map(|threads| NonZeroUsize::new(threads.parse().unwrap()).unwrap()),
         ..Config::rustc(path)
     };
+    // keep in sync with `./miri run`
+    config.comment_defaults.base().edition = Some(Spanned::dummy("2021".into())).into();
```

Comment:
It's a bit unclear what should be kept in sync.
```suggestion
    let edition = "2021"; // keep in sync with `./miri run`
    config.comment_defaults.base().edition = Some(Spanned::dummy(edition.into())).into();
```
But also... this is quite confusing, why do we need to `Spanned` everything?